### PR TITLE
minor refactoring and reuqirements.txt update

### PR DESCRIPTION
**Pull request by sahel-sh**
This cl:
1- Changes the retriever.py to always return retrieved results, rather than None in case of pyserini retriver.
2- reranker class does not need the dataset name, I added it as an arg to the write function.
3- Updated requirments.txt to flexible versions, rankzephyr needs more recent transformers version

### CODE REVIEW

**Comment by sahel-sh** [comment]
Updated script files are only format changes

File: requirements.txt [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
-transformers==4.31.0
-pyserini==0.22.0
-python-dotenv==1.0.0
-faiss-gpu==1.7.2
-faiss-cpu==1.7.2
-ftfy==6.1.1
-fschat==0.2.28
-accelerate==0.23.0
\ No newline at end of file
+transformers>=4.31.0
```
Comment (ronakice) [inline requirements.txt:4]: wouldn't this have to be a higher version minimum to accommodate mistral based models?

Comment (sahel-sh) [reply requirements.txt]: Fixed it

File: requirements.txt [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
-pyserini==0.22.0
-python-dotenv==1.0.0
-faiss-gpu==1.7.2
-faiss-cpu==1.7.2
-ftfy==6.1.1
-fschat==0.2.28
-accelerate==0.23.0
\ No newline at end of file
+tqdm>=4.66.1
+openai>=1.9.0
```
[the remark this answers is not in the document]
Comment (jasper-xian) [reply requirements.txt:2]: Also, should these changes also reflect in the pyproject.toml file?

[the remark this answers is not in the document]
Comment (sahel-sh) [reply requirements.txt:2]: you are right, I won't make the change, but I will link to this pr in #46. Thank you for catching this!

**Review by jasper-xian — APPROVED** [review]

**Review by lintool — APPROVED** [review]

**Pull request merged**